superset-frontend/plugins/plugin-chart-cartodiagram/src/thematic/components/TimeSlider.tsx

```diff
@@ -55,30 +74,86 @@ export const TimeSlider: React.FC<TimeSliderProps> = props => {
     [timesliderTooltipFormat],
   );
 
+  const advanceSlider = useCallback(() => {
+    const currentIdx = markKeys.indexOf(currentValue);
+    const nextIdx = (currentIdx + 1) % markKeys.length;
+    const nextVal = markKeys[nextIdx];
+    setCurrentValue(nextVal);
+    onChange?.(nextVal);
+  }, [currentValue, markKeys, onChange]);
+
+  const togglePlay = () => setIsPlaying(prev => !prev);
+
   useEffect(() => {
-    if (!marks || !onChange) {
-      return;
+    if (isPlaying) {
+      intervalRef.current = setInterval(advanceSlider, 1000);
+    } else {
+      if (intervalRef.current) clearInterval(intervalRef.current);
     }
-    const markKeys = Object.keys(marks);
-    if (
-      defaultValue === undefined ||
-      !markKeys.includes(defaultValue.toString())
-    ) {
-      onChange(parseInt(markKeys[0], 10));
+
+    return () => {
+      if (intervalRef.current) clearInterval(intervalRef.current);
+    };
+  }, [isPlaying, advanceSlider]);
+
+  useEffect(() => {
+    if (defaultValue === undefined || !markKeys.includes(defaultValue)) {
+      const first = markKeys[0];
+      setCurrentValue(first);
+      onChange?.(first);
     }
-  }, [defaultValue, marks, onChange]);
+  }, [defaultValue, markKeys, onChange]);
+
+  const handleSliderChange = (val: number) => {
+    setCurrentValue(val);
+    onChange?.(val);
+  };
 
   return (
-    <Slider
-      min={minVal}
-      max={maxVal}
-      marks={marks}
-      step={null}
-      tipFormatter={formatter}
-      defaultValue={defaultValue}
-      onChange={onChange}
-      {...sliderProps}
-    />
+    <div
```

Comment:
nitpick: can you use the approach suggested by emotion (and by the other components), when styling elements?